crates/biome_js_analyze/src/analyzers/nursery/no_unused_private_class_members.rs

```diff
@@ -0,0 +1,299 @@
+
+                            if is_write_only(&js_name) && !is_accessor {
+                                continue;
+                            }
+
+                            if is_update_expression(&js_name) {
+                                continue;
+                            }
+                        }
+
+                        private_members.remove(&token);
+                    }
+                }
+            }
+            biome_rowan::WalkEvent::Leave(_) => {
+                if private_members.is_empty() {
+                    break;
+                }
+            }
+        }
+    }
+
+    private_members.into_iter().map(|member| member.1).collect()
+}
+
+fn get_all_declarated_private_members(
+    class_declaration: &JsClassDeclaration,
+) -> FxHashMap<TokenText, RuleState> {
+    let constructor_params = class_declaration
+        .members()
+        .syntax()
+        .descendants()
+        .filter_map(TsPropertyParameter::cast)
+        .filter(|param| is_private(param.syntax()))
+        .filter_map(|param| {
+            let param_name = param
+                .formal_parameter()
+                .ok()?
+                .as_js_formal_parameter()?
+                .binding()
+                .ok()?
+                .as_any_js_binding()?
+                .as_js_identifier_binding()?
+                .name_token()
+                .ok()?;
+
+            let text_token = param_name.token_text_trimmed();
+            let text_range = param_name.text_trimmed_range();
+
+            let state = RuleState {
+                range: text_range,
+                member: AnyPrivateMember::TsPropertyParameter(param),
+            };
+
+            Some((text_token, state))
+        });
+
+    class_declaration
+        .members()
+        .iter()
+        .filter(|member| is_private(member.syntax()))
+        .filter_map(|member| {
+            let text_token = get_member_text_token(&member)?;
+            let text_range = get_member_range(&member)?;
+            let state = RuleState {
+                range: text_range,
+                member: AnyPrivateMember::AnyJsClassMember(member),
+            };
+
+            Some((text_token, state))
+        })
+        .chain(constructor_params)
+        .collect()
+}
+
+fn is_private(syntax: &SyntaxNode<JsLanguage>) -> bool {
+    syntax.descendants().any(|node| {
+        let is_es_private = JsPrivateClassMemberName::cast_ref(&node).is_some();
+        let is_ts_private = TsAccessibilityModifier::cast_ref(&node)
+            .is_some_and(|accessibility| accessibility.is_private());
+
+        is_es_private || is_ts_private
+    })
+}
+
+/// Check whether the js name is in a write only assignment.
+fn is_write_only(js_name: &AnyJsName) -> bool {
+    let grand_parent = js_name.syntax().grand_parent();
+    let is_assignment_expression = grand_parent
+        .kind()
+        .is_some_and(|kind| matches!(kind, JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION));
```

Comment:
You can avoid a redundant check by writing instead:

```suggestion
        .is_some_and(|grand_pararent| matches!(grand_pararent.kind(), JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION));
```

otherwise we could write:

```suggestion
        .kind()
        .is_some_and(JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION::can_cast);
```

Reply:
Cool, I didn't know there was such a method

Reply:
@Conaclos I need to import something to use JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION::can_cast?